### ISSUE TITLE
BUG: Key combinations don't work when using languages other than English.

### ISSUE THREAD

**Issue by CyberUlitka** [zsviczian/obsidian-excalidraw-plugin#1895]
I use Cyrillic in my notes, and it's very painful to switch the keyboard layout just to get access to Ctrl+Z. I guess Excalidraw checks some kind of input text instead of seeing when I press the exact button, for example, Z, which on my keyboard is the letter Я. I don't have this problem in any of the existing editors, so I guess fixing this would not be a huge problem. Thank you so much for what you're doing!

**Comment by zsviczian** [comment]
yes. that is a known issue that also affects excalidraw.com. Unfortunately, there is not much I can do about it... but if you think it is an easy solution I welcome a PR to fix it. I don't think it is that easy, but I might be wrong. Check the English layout for the physical location of the key. The same key - independent of its mapping in your layout usually works.

**Comment by SeoHyeonMyeong** [comment]
I'm using this from South Korea and experiencing the same issue with Korean input. I inspected the keydown events on **excalidraw.com** and found some useful data that might help solve this problem.

Code I used to inspect keydown events:
```js
 document.addEventListener('keydown', (e) => {
    console.log({
      key: e.key,
      code: e.code,
      keyCode: e.keyCode
    });
  });
```
 Results:
 English mode:
 `{key: 'a', code: 'KeyA', keyCode: 65}`

 Korean mode:
 `{key: 'ㅁ', code: 'KeyA', keyCode: 65}`

**Comment by zsviczian** [comment]
@SeoHyeonMyeong, this is very interesting. Thanks for sharing.

Looking at the Excalidraw components source code I can see they are deliberately using CODE and Keys in different contexts. From this I take it the configuration is deliberate. Some keys are meant to be physical locations on the keyboard, not configured meaning based on your language settings.

https://github.com/excalidraw/excalidraw/blob/4e0441eeb413ad7234f65cbb8fbb397bc715ff46/packages/excalidraw/components/shapes.tsx#L34-L38

https://github.com/excalidraw/excalidraw/blob/4e0441eeb413ad7234f65cbb8fbb397bc715ff46/packages/excalidraw/actions/actionCanvas.tsx#L172-L175

https://github.com/excalidraw/excalidraw/blob/master/packages/common/src/keys.ts

**Comment by SeoHyeonMyeong** [comment]
https://github.com/excalidraw/excalidraw/issues/3847
I found a related issue being discussed in the excalidraw repository, and this turns out to be a much more complex problem than I initially thought.

To support QWERTY, AZERTY, QWERTZ and other Latin-based layouts, we need to use `key`. However, to support non-Latin languages with QWERTY layout, we need to use `code`.
But I have no idea how we should handle non-Latin languages with non-QWERTY layouts (like Russian ЙЦУКЕН or Arabic layouts).

**Comment by KitSowka** [comment]
@SeoHyeonMyeong, most users of the `ЙЦУКЕН` layout don't associate shortcuts with Cyrillic letters. Usually, `ЙЦУКЕН` physically matches `QWERTY`, and people navigate by looking at the Latin labels. Even in software with a Russian UI, you'll almost never see `Ctrl+Ц` instead of `Ctrl+W`.
